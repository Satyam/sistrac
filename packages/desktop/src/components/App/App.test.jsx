import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';
import Connected, { App, mapStateToProps } from '.';

describe('App component', () => {
  describe('raw component', () => {
    it('should render (shallow)', () => {
      const wrapper = shallow(<App />, { disableLifecycleMethods: true });
      expect(wrapper).toMatchSnapshot();
    });
    it('should render a blank screen when there is a user', () => {
      const props = {
        usuario: { idUsuario: 20 },
        history: {},
        location: {
          pathname: '/',
        },
      };
      const wrapper = shallow(<App {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
    it('should push to /login when no user given', () => {
      const push = jest.fn();
      const props = {
        usuario: {},
        history: { push },
        location: {
          pathname: '/',
        },
      };
      mount(
        <Router>
          <App {...props} />
        </Router>,
      );
      expect(push.mock.calls.length).toBe(1);
      expect(push.mock.calls[0][0]).toBe('/login');
    });
  });
});
