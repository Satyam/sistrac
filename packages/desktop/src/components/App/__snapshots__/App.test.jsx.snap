// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component raw component should render (shallow) 1`] = `
<div
  className="App"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Sistrac
    </title>
  </HelmetWrapper>
  <Uncontrolled(Navbar)
    bsStyle="default"
    staticTop={true}
  >
    <NavbarHeader>
      <NavbarBrand>
        <a
          href="/"
        >
          Sistrac
        </a>
      </NavbarBrand>
    </NavbarHeader>
    <Nav
      bsClass="nav"
      justified={false}
      pullLeft={false}
      pullRight={false}
      stacked={false}
    >
      <NavItem
        eventKey={1}
        href="/estaciones"
      >
        Estaciones
      </NavItem>
      <NavItem
        eventKey={2}
        href="#"
      >
        Link
      </NavItem>
      <NavDropdown
        eventKey={3}
        id="basic-nav-dropdown"
        title="Dropdown"
      >
        <MenuItem
          bsClass="dropdown"
          disabled={false}
          divider={false}
          eventKey={3.1}
          header={false}
        >
          Action
        </MenuItem>
        <MenuItem
          bsClass="dropdown"
          disabled={false}
          divider={false}
          eventKey={3.2}
          header={false}
        >
          Another action
        </MenuItem>
        <MenuItem
          bsClass="dropdown"
          disabled={false}
          divider={false}
          eventKey={3.3}
          header={false}
        >
          Something else here
        </MenuItem>
        <MenuItem
          bsClass="dropdown"
          disabled={false}
          divider={true}
          header={false}
        />
        <MenuItem
          bsClass="dropdown"
          disabled={false}
          divider={false}
          eventKey={3.4}
          header={false}
        >
          Separated link
        </MenuItem>
      </NavDropdown>
    </Nav>
  </Uncontrolled(Navbar)>
  <Route
    component={[Function]}
    path="/estaciones"
  />
  <Route
    component={[Function]}
    path="/estacion/:idEstacion"
  />
  <Route
    component={[Function]}
    path="/login"
  />
</div>
`;

exports[`App component raw component should render a blank screen when there is a user 1`] = `
<div
  className="App"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Sistrac
    </title>
  </HelmetWrapper>
  <Uncontrolled(Navbar)
    bsStyle="default"
    staticTop={true}
  >
    <NavbarHeader>
      <NavbarBrand>
        <a
          href="/"
        >
          Sistrac
        </a>
      </NavbarBrand>
    </NavbarHeader>
    <Nav
      bsClass="nav"
      justified={false}
      pullLeft={false}
      pullRight={false}
      stacked={false}
    >
      <NavItem
        eventKey={1}
        href="/estaciones"
      >
        Estaciones
      </NavItem>
      <NavItem
        eventKey={2}
        href="#"
      >
        Link
      </NavItem>
      <NavDropdown
        eventKey={3}
        id="basic-nav-dropdown"
        title="Dropdown"
      >
        <MenuItem
          bsClass="dropdown"
          disabled={false}
          divider={false}
          eventKey={3.1}
          header={false}
        >
          Action
        </MenuItem>
        <MenuItem
          bsClass="dropdown"
          disabled={false}
          divider={false}
          eventKey={3.2}
          header={false}
        >
          Another action
        </MenuItem>
        <MenuItem
          bsClass="dropdown"
          disabled={false}
          divider={false}
          eventKey={3.3}
          header={false}
        >
          Something else here
        </MenuItem>
        <MenuItem
          bsClass="dropdown"
          disabled={false}
          divider={true}
          header={false}
        />
        <MenuItem
          bsClass="dropdown"
          disabled={false}
          divider={false}
          eventKey={3.4}
          header={false}
        >
          Separated link
        </MenuItem>
      </NavDropdown>
    </Nav>
  </Uncontrolled(Navbar)>
  <Route
    component={[Function]}
    path="/estaciones"
  />
  <Route
    component={[Function]}
    path="/estacion/:idEstacion"
  />
  <Route
    component={[Function]}
    path="/login"
  />
</div>
`;
